### ISSUE TITLE
window reference in v3 causes issues with ssr

### ISSUE THREAD

**Issue by shaezzy** [Learus/react-material-ui-carousel#129]
In v3 version, the compiled js includes a window reference

```js
var divRef = (0, react_1.useRef)(null);
var handleResize = (0, react_1.useCallback)(function () {
    if (divRef.current && index === active)
        setHeight(divRef.current.clientHeight);
}, [divRef, setHeight, index, active]);
(0, react_1.useEffect)(function () {
    handleResize();
}, []);
window.addEventListener('resize', handleResize);
```

Although, it can be worked around, It would be nice if v3 also worked with SSR

**Comment by renanfcamara** [comment]
Just got this issue too, yeap... lets find another one

**Comment by hughjdavey** [comment]
@shaezzy How can it be worked around?

**Comment by Learus** [comment]
I will attempt to fix this by adding a check for `window` existence before adding the event listener.
Since using SSR actually renders the Carousel twice, it should work correctly. Specifically, I'm adding;
```tsx
// Do this check for SSR cases
if (!(typeof window === 'undefined'))
    window.addEventListener('resize', handleResize);
```

I don't have a testing environment set up for SSR, so it would be appreciated if a sandbox could be made to test this feature.

I will upload the fix in the new version (`^3.1.1`). Stay tuned!

**Comment by Learus** [comment]
Try it out now.
